### ISSUE TITLE
LuaTeX support

### ISSUE THREAD

**Issue by hmenke** [pgf-tikz/pgf-umlcd#2]
**Brief outline of the bug**

`pgf-umlcd` doesn't work with LuaTeX because of
```
! LaTeX Error: Command \attribute already defined.
               Or name \end... illegal, see p.192 of the manual.
```

**Minimal working example (MWE)**

```latex
\documentclass{article}
\usepackage{pgf-umlcd}
\begin{document}
\end{document}
```

**Comment by atticus-sullivan** [comment]
As can be seen at [luatex-doc](http://mirrors.ibiblio.org/CTAN/systems/doc/luatex/luatex.pdf)(Page 24, section 2.3.2), the macro `\attribute` is already in use by LuaTeX (for some new ~counter implementation stuff). I think only [Line 227 in the .sty](https://github.com/pgf-tikz/pgf-umlcd/blob/91e6e52387407259c1413b097f483685b89df930/tex/latex/pgf-umlcd/pgf-umlcd.sty#L227) would need to be changed for resolving. As this is the macro for the interface to the outside, the question is how to name it (`classAttribute` isn't that accurate I think).

Most important is that by simply renaming the macro there wouldn't be any backwards compatibility. I'd suggest to add the `ifluatex`/`iftex` package or some equivalent so that the new macro can be used in all versions, but prevent the redefinition of `attribute` in a lua(la)tex environment.

[Because of the open question regarding the naming and since I didn't dug into the doc code yet (might be subject to change as well), I didn't do a pull request. In addition I'm not sure if this is the right way of contributing to this project (or if there has to be some mailinglist action or something else)]

**Comment by TeXhackse** [comment]
@hmenke Is there any plan to fix this one?

I guess the only stable way to resolve this permanently would be changing the user Syntax (probably backwards compatible for pdfLaTeX). I could do such changes if that would help.  Or are there any other ideas?

**Comment by complanar** [comment]
Seems, this is not as complicated as it seems. Quick dirty fix:

Changing [Line 227 in the .sty](https://github.com/pgf-tikz/pgf-umlcd/blob/91e6e52387407259c1413b097f483685b89df930/tex/latex/pgf-umlcd/pgf-umlcd.sty#L227) to `\newcommand{\umlAttribute}[1]{` and then redefining `\attribute` only inside of the `classAndInterfaceCommon` environment by adding `\let\attribute\umlAttribute` (around Line 199) does the trick.

I'm only a LaTeX user and no package author, so this is maybe dirty code and does not meet the standards, but probably someone knows a better solution.
In the meantime I kind of “fixed“ my copy of the package this way. As of now I didn't experience any side effects.

**Comment by TeXhackse** [comment]
I currently use the hook management to workaround it, so `\attribute`  is overwritten within the internal environments of this package. But it's not a good solution to overwrite it. There might be a usecase which might make use of the luatex attribute within the environment. So it's not a good solution to overwrite it. Not even locally.

**Comment by complanar** [comment]
You’re right, that‘s why I called it a quick dirty fix. I didn’t want to recommend this solution for the package maintainer, but for me it is better than not to be able to use the package at all. Just wanted to share this workaround with others.
